resolver/resolver.go

```diff
@@ -105,8 +109,22 @@ type Address struct {
 // BuildOption includes additional information for the builder to create
 // the resolver.
 type BuildOption struct {
-	// DisableServiceConfig indicates whether resolver should fetch service config data.
+	// DisableServiceConfig indicates whether a resolver implementation should
+	// fetch service config data.
 	DisableServiceConfig bool
+	// DialCreds is the transport credentials that a resolver implementation
+	// can use to dial a remote name resolution server. Resolver
```

Comment:
Maybe this should say simply that it's the DialCreds for the ClientConn?  It may not be appropriate to use these credentials for your name resolver in many (practically all?) cases.  Same for the other two fields, except that a Bundle is a little more likely to be OK to use.  Possibly include a warning about using this.

Reply:
Does this sound OK? If so, I can make similar modifications to the other two fields as well.

```
DialCreds is the transport credentials used by the ClientConn (set as a DialOption
using a call to WithTransportCredentials) while communicating with the gRPC
server. In some cases, it may be appropriate for a resolver implementation to use 
these credentials to talk to a remote name resolution server. In most cases though, 
it might not be appropriate and should be used with caution.
```